wireless_modules/wireless_module.py

```diff
@@ -38,6 +42,22 @@ def __init__(self, module_id, battery_reader=None):
         self.mqtt = Client(client_id, config.MQTT_BROKER)
 
         self.battery = battery_reader
+        self.status_led = status_led
+
+        asyncio.get_event_loop().set_exception_handler(self.error_handler)
+
+    def error_handler(self, _loop, context):
+        self.status_led.set_state(WmState.Error)
+        print("An error occured in a uasyncio task!")
+        print(context["message"])
+        sys.print_exception(context["exception"])
+
+        # FIXME: A nicer solution would be to restart all running tasks rather than reset
```

Comment:
Is there an easy way to gracefully stop all tasks before reseting the esp32?

Reply:
Ordinarily there would be, but asyncio doesn't let you "cancel" a task from within the task itself. This function is considered by asyncio to be part of the task that the error occurred in, so unfortunately there's not a simple clean way to cancel the tasks.

To be honest, there's no harm in resetting the esp32 without notice anyway, unlike for example a raspberry pi which mightn't be as big of a fan. When we turn off the esp32 we don't gracefully stop tasks either, we just cut power.

The main reason it would be good to restart tasks rather than reset would be that we could come back online faster and maintain state such as whether we should be logging. However, if it's something we decide we want I think it could go in its own PR, kinda unrelated to the status LED at this point. Another solution to the problem would be to just make the `start`/`stop` topics retained (I reckon we should)

Reply:
Yep agree `start/stop` should be retained and completely fine with the reset task to be a seperate PR. I didn't see any benefit in gracefully stopping either, it just seemed nice 😂. Hence my initial emphasise on whether `there is an easy way` to do it.